mshadow/base.h

```diff
@@ -877,6 +888,44 @@ struct minimum {
     LOG(FATAL) << "Unknown type enum " << type;     \
   }
 
+#define MSHADOW_FLOAT_DOUBLE_TYPE_SWITCH(type, DType, ...)  \
```

Comment:
Is the intent here to simply error on anything except float and double?
I don't see it used in this diff?